service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/documents/ManageDocumentsLAController.java

```diff
@@ -173,6 +178,29 @@ public AboutToStartOrSubmitCallbackResponse handleFurtherEvidenceMidEvent(@Reque
         return respond(caseDetails);
     }
 
+    @PostMapping("/final-check/mid-event")
+    @java.lang.SuppressWarnings("java:S5852")
+    public AboutToStartOrSubmitCallbackResponse handleFinalCheckMidEvent(@RequestBody CallbackRequest request) {
+        CaseDetails caseDetails = request.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        List<String> errors = new ArrayList<>();
+
+        if (FURTHER_EVIDENCE_DOCUMENTS.equals(caseData.getManageDocumentLA().getType())) {
+            List<Element<ApplicationDocument>> swetDocs = caseData.getApplicationDocuments().stream()
+                .filter(df -> !StringUtils.isEmpty(df.getValue().getIncludedInSWET()))
+                .collect(Collectors.toList());
+            if (!swetDocs.isEmpty()) {
+                Pattern pattern = Pattern.compile("(?s)(?!.*<[^>\\d\\n]+>*).*");
```

Comment:
This definitely doesn't work on the ccd config file?

Reply:
I have tried that in CCD configuration but it doesn't support it. I have checked the ccd code it does use java to compile the regular expression defined in CCD config. It works fine. However, it seems that the ExUI will parse the regular expression as well in javascript and it breaks the UI stuff showing a blank page.

Thus, for a workaround, I put the validation logic in the mid event.